sdk/core/azure-core/src/main/java/com/azure/core/http/rest/RestProxy.java

```diff
@@ -83,17 +83,15 @@ public Object invoke(Object proxy, final Method method, Object[] args) {
         final SwaggerMethodParser methodParser = getMethodParser(method);
         RequestOptions options = methodParser.setRequestOptions(args);
         Context context = methodParser.setContext(args);
-        boolean isReactive = methodParser.isReactive();
-        boolean syncRestProxyEnabled = (boolean) context.getData(HTTP_REST_PROXY_SYNC_PROXY_ENABLE)
-            .orElse(GLOBAL_SYNC_PROXY_ENABLE);
+        boolean syncRestProxyEnabled = (boolean) context.getData(HTTP_REST_PROXY_SYNC_PROXY_ENABLED)
+            .orElse(GLOBAL_SYNC_PROXY_ENABLED);
```

Comment:
Let's move this down into the else block below and make this calling more verbose.

```java
if (isReactive) {
  // async call
} else {
  if (syncDisabled) {
    // blocking async call
  } else {
    // full sync call
  }
}
```

That way we only check the sync proxy enabled context when we actually have to instead of always checking it even when it won't be used (`methodParser.isReactive() == true`)